From ca576e0804324bbb36b697543cbe992e34d4b951 Mon Sep 17 00:00:00 2001
From: Eike Rathke <erack@redhat.com>
Date: Mon, 7 Oct 2013 21:51:26 +0200
Subject: [PATCH] WaE [-Wstrict-overflow] assuming signed overflow does not
 occur

... when assuming that (X - c) <= X is always true
... or that (X + c) < X is always false

Change-Id: Ib2313827cd6358ced0141b41cba753896b676e28
---
 svtools/source/contnr/imivctl1.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/svtools/source/contnr/imivctl1.cxx b/svtools/source/contnr/imivctl1.cxx
index 1cd006b..0685bc3 100644
--- a/svtools/source/contnr/imivctl1.cxx
+++ b/svtools/source/contnr/imivctl1.cxx
@@ -2710,9 +2710,9 @@ Rectangle SvxIconChoiceCtrl_Impl::CalcFocusRect( SvxIconChoiceCtrlEntry* pEntry
     Rectangle aFocusRect( aBoundRect.Left(), aBmpRect.Top() - 1,
                           aBoundRect.Right() - 4, aTextRect.Bottom() + 1 );
     // the focus rectangle should not touch the text
-    if( aFocusRect.Left() - 1 >= pEntry->aRect.Left() )
+    if( aFocusRect.Left() > ::std::numeric_limits<long>::min() && aFocusRect.Left() - 1 >= pEntry->aRect.Left() )
         aFocusRect.Left()--;
-    if( aFocusRect.Right() + 1 <= pEntry->aRect.Right() )
+    if( aFocusRect.Right() < ::std::numeric_limits<long>::max() && aFocusRect.Right() + 1 <= pEntry->aRect.Right() )
         aFocusRect.Right()++;
 
     return aFocusRect;
-- 
1.8.3.1

